tests/components/blue_current/test_init.py

```diff
@@ -0,0 +1,185 @@
+"""Test Blue Current Init Component."""
+
+from datetime import timedelta
+from unittest.mock import patch
+
+from bluecurrent_api.client import Client
+from bluecurrent_api.exceptions import RequestLimitReached, WebsocketError
+import pytest
+
+from homeassistant.components.blue_current import DOMAIN, Connector, async_setup_entry
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from . import init_integration
+
+from tests.common import MockConfigEntry
+
+
+async def test_load_unload_entry(hass: HomeAssistant) -> None:
+    """Test load and unload entry."""
+    config_entry = await init_integration(hass, "sensor", {})
+    assert config_entry.state == ConfigEntryState.LOADED
+    assert isinstance(hass.data[DOMAIN][config_entry.entry_id], Connector)
+
+    await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert hass.data[DOMAIN] == {}
+
+
+async def test_config_not_ready(hass: HomeAssistant) -> None:
+    """Tests if ConfigEntryNotReady is raised when connect raises a WebsocketError."""
+    with patch(
+        "bluecurrent_api.Client.connect",
+        side_effect=WebsocketError,
+    ), pytest.raises(ConfigEntryNotReady):
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id="uuid",
+            unique_id="uuid",
+            data={"api_token": "123", "card": {"123"}},
+        )
+        config_entry.add_to_hass(hass)
+
+        await async_setup_entry(hass, config_entry)
+
+
+async def test_on_data(hass: HomeAssistant) -> None:
+    """Test on_data."""
+
+    await init_integration(hass, "sensor", {})
+
+    with patch(
+        "homeassistant.components.blue_current.async_dispatcher_send"
```

Comment:
Don't patch integration details. Patch the library and assert calls to the library and entity state in the state machine.